### PR TITLE
(refactor) Switch form launcher workspace to OpenMRS forms workspace

### PR DESCRIPTION
**Pull request by CynthiaKamau**
…s workspace

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary

- This pr entails removing the custom react form engine workspace created and reusing the openmrs forms workspace. With this, users can launch the angular/react forms from the ohri clinical views freely
- Required form uuid to be passed to the workspace,
- Read only mode breaks for now and PMTCT antenatal form is missing

https://github.com/UCSF-IGHS/openmrs-esm-ohri/assets/19533785/ce43bce6-7227-46e0-b867-695faff3c21c

### CODE REVIEW

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 1]
```diff
@@ -90,6 +92,11 @@ export const EncounterList: React.FC<EncounterListProps> = ({
     [forms, t],
   );
 
+  console.log("---forms", forms)
+  console.log("---forsJson", formsJson, isLoadingFormsJson)
+  console.log("---formlist", formList)
```
Comment (kajambiya) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:97]: Hi, you might want to remove all the console logs.

File: packages/esm-commons-lib/src/components/encounter-list/helpers.ts [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
-    moduleName,
-    title || defaultTitle,
-    encounterUuid,
-    intent,
-    onFormSave,
-    workspaceWindowSize,
-  );
+  const defaultTitle = capitalize(action) + ' ' + form?.name;
+  console.log("---helper form", form)
+  console.log("---helper intent", intent)
```
Comment (kajambiya) [inline packages/esm-commons-lib/src/components/encounter-list/helpers.ts:21]: Remove these console logs as well

**Review by kajambiya — COMMENTED** [review]
Hi @CynthiaKamau Nice work.
Just a few comments. 
1. I see you forgot to do some clean up; you left lots of console logs.
2. You might want to pass the form uuids in a config file because another implementer will have a different uuid of the same form.
3. What happens if the `formList` has more than one form?
4. I see a failed e2e test. Don't know whether it's @lucyjemutai's issue to fix.

File: packages/esm-commons-lib/src/components/encounter-list/helpers.ts (hunk 2 above)
Comment (kajambiya) [reply packages/esm-commons-lib/src/components/encounter-list/helpers.ts:21]: Ignore the console log comments. Hadn't noticed that it was a draft PR

**Comment by CynthiaKamau** [comment]
> Hi @CynthiaKamau Nice work. Just a few comments.
> 
> 1. I see you forgot to do some clean up; you left lots of console logs.
> 2. You might want to pass the form uuids in a config file because another implementer will have a different uuid of the same form.
> 3. What happens if the `formList` has more than one form?
> 4. I see a failed e2e test. Don't know whether it's @lucyjemutai's issue to fix.

Thanks @kajambiya , ive moved the uuids to the config. From the previous implementation, if a form List has more than one form, the first one was picked

**Comment by pirupius** [comment]
@CynthiaKamau the video seems to be cropped. 
1.  Was the white screen a manual refresh? 
2. Is there some work that needs to be added in mutating the encounter list data?

File: packages/esm-cervical-cancer-app/src/views/cacx-visits/tabs/cacx-treatment.component.tsx [hunk 3]
```diff
@@ -100,7 +102,7 @@ export const CacxTreatment: React.FC<CacxTreatmentProps> = ({ patientUuid }) =>
     <EncounterList
       patientUuid={patientUuid}
       encounterType={cacxTreatmentEncounterType_UUID}
-      formList={[{ name: screeningAndCancerTreatmentForm }]}
+      formList={[{ name: screeningAndCancerTreatmentForm, uuid: screeningAndCancerTreatmentFormUuid }]}
```
[the remark this answers is not in the document]
Comment (CynthiaKamau) [reply packages/esm-cervical-cancer-app/src/views/cacx-visits/tabs/cacx-treatment.component.tsx:105]: I am using the workspace registered on the openmrs react form engine workspace, with that, users can also use ohri clinical views even if they use the angular form engine. Nothing has changed on the backend, the formUuid is a required prop for that workspace

File: packages/esm-commons-lib/src/components/encounter-list/helpers.ts [hunk 4]
```diff
@@ -1,28 +1,33 @@
-import { OHRIFormSchema, SessionMode } from '@openmrs/openmrs-form-engine-lib';
-import { launchForm } from '../../utils/ohri-forms-commons';
-import { capitalize } from 'lodash-es';
+import { launchPatientWorkspace } from '@openmrs/esm-patient-common-lib';
 
 type LaunchAction = 'add' | 'view' | 'edit';
 
 export function launchEncounterForm(
-  form: OHRIFormSchema,
+  form: any,
```
[the remark this answers is not in the document]
Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/components/encounter-list/helpers.ts:6]: I can work on updating that type

File: packages/esm-covid-app/src/views/case-assessment.encounter-lists.tsx [hunk 5]
```diff
@@ -104,9 +106,17 @@ const CovidAssessment: React.FC<CovidAssessmentWidgetProps> = ({ patientUuid })
       patientUuid={patientUuid}
       encounterType={config.encounterTypes.covid_Assessment_EncounterUUID}
       formList={[
-        { name: config.formNames.CovidAssessmentFormName, excludedIntents: ['COVID_LAB_ASSESSMENT_EMBED'] },
-        { name: config.formNames.CovidCaseFormName },
-        { name: config.formNames.CovidOutcomeFormName, excludedIntents: ['COVID_OUTCOME_EMBED', '*'] },
+        {
+          name: config.formNames.CovidAssessmentFormName,
+          excludedIntents: ['COVID_LAB_ASSESSMENT_EMBED'],
+          uuid: covidAssessmentFormUuid,
```
[the remark this answers is not in the document]
Comment (CynthiaKamau) [reply packages/esm-covid-app/src/views/case-assessment.encounter-lists.tsx:112]: Im trying to reduce maintenance on our side being that the current workspace is out of date and updating it has some few challenges

File: packages/esm-cervical-cancer-app/src/views/cacx-visits/tabs/cacx-treatment.component.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (pirupius) [reply packages/esm-cervical-cancer-app/src/views/cacx-visits/tabs/cacx-treatment.component.tsx:105]: We need to change that. My guess is that code was written before we moved to using the name.

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 6]
```diff
@@ -188,14 +192,15 @@ export const EncounterList: React.FC<EncounterListProps> = ({
                 onClick={(e) => {
                   e.preventDefault();
                   launchEncounterForm(
-                    forms.find((form) => form.name == actionItem?.form?.name),
+                    formList[0],
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:195]: Why did u change this?  On certain scenarios, actions can have more than one form and that's why it was done this way. You could take a look at covid assessment

File: packages/esm-commons-lib/src/components/encounter-list/helpers.ts (hunk 4 above)
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/helpers.ts:6]: The `OHRIFormSchema` comes from the form engine. Are there any missing values that necessitated this change?

**Comment by CynthiaKamau** [comment]
> @CynthiaKamau the video seems to be cropped.
> 
> 1. Was the white screen a manual refresh?
> 2. Is there some work that needs to be added in mutating the encounter list data?

That can be handled in a follow up pr

File: packages/esm-commons-lib/src/components/encounter-list/helpers.ts (hunk 4 above)
Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/components/encounter-list/helpers.ts:6]: yes, but i can look into finding a workaround

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx (hunk 6 above)
Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:195]: I still don't get this, how does launching more than one form work? I thought the ohri form component only accepts one form uuid

**Review by kajambiya — COMMENTED** [review]
LGTM

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 7]
```diff
@@ -83,7 +84,7 @@ export const EncounterList: React.FC<EncounterListProps> = ({
         form: {
           name: forms[0]?.name,
         },
-        mode: 'view',
+        mode: 'edit',
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:87]: This looks scary 👀 how was this working with the wrong mode

File: packages/esm-cervical-cancer-app/src/views/cacx-visits/tabs/cacx-treatment.component.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (pirupius) [reply packages/esm-cervical-cancer-app/src/views/cacx-visits/tabs/cacx-treatment.component.tsx:105]: Circling back to this, did u mean it's required in the `form-engine-app` within the patient chart

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 8]
```diff
@@ -145,6 +146,7 @@ export const EncounterList: React.FC<EncounterListProps> = ({
               encounter.uuid,
               null,
               workspaceWindowSize,
+              patientUuid,
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:149]: Am still skeptical about breaking changes. Don't we have an instance of the patient in the workspace? Not sure we need this.

Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:149]: https://github.com/openmrs/openmrs-esm-patient-chart/blob/6d0b682a14a2265d38d3544a9d54f0e5672d9ae7/packages/esm-patient-common-lib/src/form-entry-interop.ts#L55

**Review by pirupius — APPROVED** [review]

**Pull request merged**